src/main/scala/org/powerapi/module/libpfm/LibpfmCoreProcessSensor.scala

```diff
@@ -0,0 +1,153 @@
+ *
+ * This file is a part of PowerAPI.
+ *
+ * Copyright (C) 2011-2014 Inria, University of Lille 1.
+ *
+ * PowerAPI is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version.
+ *
+ * PowerAPI is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with PowerAPI.
+ *
+ * If not, please consult http://www.gnu.org/licenses/agpl-3.0.html.
+ */
+package org.powerapi.module.libpfm
+
+import java.util.{BitSet, UUID}
+import akka.util.Timeout
+import org.powerapi.core.{OSHelper, APIComponent, MessageBus, Target}
+
+/**
+ * Main actor for getting the performance counter value per core/event/process.
+ *
+ * @author Maxime Colmant <maxime.colmant@gmail.com>
+ */
+class LibpfmCoreProcessSensor(eventBus: MessageBus, timeout: Timeout, osHelper: OSHelper, configuration: BitSet, events: Array[String], cores: Map[Int, List[Int]], inDepth: Boolean) extends APIComponent {
+  import akka.actor.{Actor, ActorIdentity, Identify, PoisonPill, Props}
+  import akka.event.LoggingReceive
+  import akka.pattern.ask
+  import org.powerapi.core.{Application, Process}
+  import org.powerapi.core.MonitorChannel.{MonitorTick, subscribeMonitorTick}
+  import org.powerapi.module.SensorChannel.{MonitorStop, MonitorStopAll, subscribeSensorsChannel}
+  import PerformanceCounterChannel.{formatLibpfmCoreProcessSensorChildName, PCWrapper, publishPCReport}
+  import scala.concurrent.{Await, Future}
+  import scala.reflect.ClassTag
+
+  override def preStart(): Unit = {
+    subscribeMonitorTick(eventBus)(self)
+    subscribeSensorsChannel(eventBus)(self)
+    super.preStart()
+  }
+
+  val processClaz = implicitly[ClassTag[Process]].runtimeClass
+  val appClaz = implicitly[ClassTag[Application]].runtimeClass
+
+  def receive: Actor.Receive = running(Map(), Map(), Map())
+
+  def running(targets: Map[UUID, Set[Target]], timestamps: Map[UUID, Long], identifiers: Map[(UUID, Target), Set[Int]]): Actor.Receive = LoggingReceive {
+    case monitorTick: MonitorTick if processClaz.isInstance(monitorTick.target) || appClaz.isInstance(monitorTick.target) => {
+      sense(monitorTick, targets, timestamps, identifiers)
+    }
+    case msg: MonitorStop => monitorStopped(msg)
+    case msg: MonitorStopAll => monitorAllStopped(msg)
+  } orElse default
+
+  def sense(monitorTick: MonitorTick, targets: Map[UUID, Set[Target]], timestamps: Map[UUID, Long], identifiers: Map[(UUID, Target), Set[Int]]): Unit = {
+    var wrappers = Map[(Int, String), PCWrapper]()
+
+    var _targets = targets
+    var _timestamps = timestamps
```

Comment:
Why do you allocate these three variables instead of working with the parameters?